### PR TITLE
Add support for identity columns

### LINKED ISSUE #159: Support identity columns
Add support to the dialect and alembic

### PR DESCRIPTION
**Pull request by kasium**
Closes #159

### CODE REVIEW

File: sqlalchemy_hana/dialect.py [hunk 1]
```diff
@@ -436,12 +438,25 @@ def visit_drop_view(self, drop: DropView | BaseDropView, **kw: Any) -> str:
...
+        return super().visit_create_column(create, first_pk, **kw)
+
 
 class HANAExecutionContext(default.DefaultExecutionContext):
     def fire_sequence(self, seq: Sequence, type_: Integer) -> int:
         seq = self.identifier_preparer.format_sequence(seq)
         return self._execute_scalar(f"SELECT {seq}.NEXTVAL FROM DUMMY", type_)
 
+    def get_lastrowid(self) -> int:
+        self.cursor.execute("SELECT CURRENT_IDENTITY_VALUE () FROM DUMMY")
```
Comment (jarus) [inline sqlalchemy_hana/dialect.py:455]:
```suggestion
        self.cursor.execute("SELECT CURRENT_IDENTITY_VALUE() FROM DUMMY")
```

File: sqlalchemy_hana/dialect.py [hunk 2]
```diff
@@ -436,12 +438,25 @@ def visit_drop_view(self, drop: DropView | BaseDropView, **kw: Any) -> str:
...
+        res = self.cursor.fetchone()
+        assert res, "No lastrowid available"
```
Comment (jarus) [inline sqlalchemy_hana/dialect.py:457]: Is an Assertion Error expected or could we raise a proper exception?

File: sqlalchemy_hana/dialect.py [hunk 3]
```diff
@@ -475,14 +490,14 @@ class HANAHDBCLIDialect(default.DefaultDialect):
 
     div_is_floordiv = False
     implicit_returning = False
-    postfetch_lastrowid = False
+    postfetch_lastrowid = True
```
Comment (jarus) [inline sqlalchemy_hana/dialect.py:493]: Has this any kind of implications on tables with a custom sequence?

**Review by jarus — CHANGES_REQUESTED** [review]

File: sqlalchemy_hana/dialect.py (hunk 2 above)
Comment (kasium) [reply sqlalchemy_hana/dialect.py:457]: I think this should never really happen, but the assert is for us and mypy

File: sqlalchemy_hana/dialect.py (hunk 3 above)
Comment (kasium) [reply sqlalchemy_hana/dialect.py:493]: Nope, if you have a sequence, no lasrwoid is used at all. This is verified AFAIK by the SQLAlchemy tests

**Review by jarus — APPROVED** [review]

**Pull request merged**